crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -919,20 +920,20 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientState {
                 if let (Some(focused_window), Some(mouse_location)) =
                     (focused_window, mouse_location)
                 {
-                    let value = value * state.scroll_direction;
+                    let value = SCROLL_MULTIPLIER * value.signum() * state.scroll_direction;
                     focused_window.handle_input(PlatformInput::ScrollWheel(ScrollWheelEvent {
                         position: *mouse_location,
                         delta: match axis {
                             wl_pointer::Axis::VerticalScroll => {
-                                ScrollDelta::Pixels(Point::new(Pixels(0.0), Pixels(value as f32)))
```

Comment:
Doing both a scroll multiplier and reporting these, unconditionally, as line offsets seems intuitively wrong.

Either this should always report precise scrolling (as it [seems](https://github.com/kovidgoyal/kitty/issues/4694#issuecomment-1036984245) that wayland does), or we should do proper detection of whether this number is line or pixel based.

Reply:
I think I named the variable wrong, it's not really a scroll multiplier. It is the amount of lines to scroll for each scroll event. (it is multiplied by the *sign* of the received `value` to get the correct scroll direction.)

Precise scrolling should be supported for devices like touchpads, but I don't have a laptop to test a implementation. It seems that Chromium [unconditionally](https://github.com/chromium/chromium/blob/main/ui/ozone/platform/wayland/host/wayland_pointer.cc#L189-L219) (?) multiplies the value, but maybe there is a different codepath for touchpads?

Sorry if I misunderstand what you're saying, English is not my first language... :sweat_smile:

Reply:
Hello, I just wanted to say that I tested this PR on my laptop under GNOME Wayland, and the scrolling is very very fast. Currently (in `main`) the touchpad scrolling has a very usable speed, even if it lacks an inertia/kinetic effect for now

Reply:
> Hello, I just wanted to say that I tested this PR on my laptop under GNOME Wayland, and the scrolling is very very fast. Currently (in `main`) the touchpad scrolling has a very usable speed, even if it lacks an inertia/kinetic effect for now

Right now though, the scrolling is basically unusable with an actual physical scroll wheel.

Reply:
> For scroll events (vertical and horizontal scroll axes), the value parameter is the length of a vector along the specified axis in a coordinate space identical to those of motion events, representing a relative movement along the specified axis.
> -- https://wayland.app/protocols/wayland#wl_pointer:event:axis

My initial understanding of this would be that the value is a fixed-point value representing a percentage of the surface size along the given axis? I.e. if it's multiplied by the pixel size of the surface you get scroll delta in pixels?

Reply:
The [wayland-rs docs](https://smithay.github.io/wayland-rs/wayland_client/protocol/wl_pointer/enum.Event.html#variant.Axis) do make it sound sound more like a straight pixel delta.